test/integration/error_messages_test.cpp

```diff
@@ -0,0 +1,54 @@
+#include "specexamples.h"
+#include "yaml-cpp/yaml.h"  // IWYU pragma: keep
+
+#include "gtest/gtest.h"
+
+#define EXPECT_THROW_EXCEPTION(exception_type, statement, message) \
+  ASSERT_THROW(statement, exception_type);                         \
+  try {                                                            \
+    statement;                                                     \
+  } catch (const exception_type& e) {                              \
+    EXPECT_EQ(e.msg, message);                                     \
+  }
+
+namespace YAML {
+namespace {
+
+TEST(ErrorMessageTest, Ex9_1_BadSubscriptErrorMessage) {
```

Comment:
Inline the test's YAML for these specific feature tests rather than reusing the spec. (It's harder to understand when you have to go dig for these.)